src/ipahealthcheck/dogtag/range.py

```diff
@@ -0,0 +1,170 @@
+            return
+
+        yield Result(self, constants.SUCCESS,
+                     key='nextRange',
+                     nextrange=nextrange)
+
+        filterstr = "(objectclass=pkiRange)"
+        attrlist = ['beginRange', 'endRange', 'host']
+        try:
+            entries = self.conn.get_entries(
+                DN('ou=requests,ou=ranges,o=ipaca'),
+                ldap.SCOPE_SUBTREE, filterstr, attrlist)
+        except errors.EmptyResult:
+            entries = []
+
+        ranges = []
+        if entries:
+            for entry in entries:
+                host = entry.single_value.get('host')
+                msg = None
+                beginrange = int(entry.single_value.get('beginRange', 0))
+                endrange = int(entry.single_value.get('endRange', 0))
+                if beginrange == 0:
+                    msg = 'beginRange is not defined'
+                if endrange == 0:
+                    msg = 'endRange is not defined'
+                if endrange < beginrange:
+                    msg = 'endRange is less than beginRange'
+                if endrange > nextrange:
+                    msg = 'endRange is greater than nextRange'
+                if msg:
+                    yield Result(self, constants.ERROR,
+                                 key=host,
+                                 beginrange=beginrange,
+                                 endrange=endrange,
+                                 msg=msg)
+                    continue
+                ranges.append((beginrange, endrange))
+
+                yield Result(self, constants.SUCCESS,
+                             key=host,
+                             beginrange=beginrange,
+                             endrange=endrange)
+
+            range_len = len(ranges)
+            for i in range(range_len - 1):
+                for j in range(i+1, range_len):
+                    if range_overlap(ranges[i], ranges[j]):
+                        yield Result(self, constants.ERROR,
+                                     key=entries[i].single_value.get('host'),
+                                     beginrange=ranges[i][0],
+                                     endrange=ranges[i][1],
+                                     msg='Range overlap')
+
+
+@registry
+class DogtagConfigRangeCheck(DogtagPlugin):
+    """
+    Report on current ranges set in CS.cfg.
+
+    This is for information purposes only. An external tool will need
+    to combine all the values and evaluate for overlap or other issues.
+    """
+    @duration
+    def check(self):
+
+        if not self.ca.is_configured():
+            logger.debug("No CA configured, skipping dogtag config check")
+            return
+
+        # (key_name, required)
+        keys = [('dbs.beginSerialNumber', True),
+                ('dbs.endSerialNumber', True),
+                ('dbs.nextBeginSerialNumber', False),
+                ('dbs.nextEndSerialNumber', False)]
```

Comment:
And yeah, same comment here, this looks fine for serials, but we need to do similar thing for all the other managed ranges.